newIDE/app/src/MainFrame/ElectronMainMenu.js

```diff
@@ -145,6 +156,17 @@ class ElectronMainMenu extends React.Component<Props, {||}> {
     }
   }
 
+  componentWillMount() {
```

Comment:
componentWillMount would only generate the menu when the component is first shown, i.e: the app startup. It would not update if we open a new file. 
To fix this, we'll rely on the fact that at any time to change the recent file, then the PreferencesProvider is doing a setState. 
Which then re-renders any consumer of the context... including the one that gives the props to MainFrame and ElectronMainMenu :) 
Then in ElectronMainMenu, we still don't want to always re-render the menu (in fact, most of the case we don't want to update it) - as it might be a bit costly. But here, we want to trigger an update to the menu.
How?
The component will be re-render when the props are updated. `render` will be called (we do nothing special) and then `componentDidUpdate`, if any, will be called.
At this moment, we can do something in the `componentDidUpdate`: we compare the result of `this.props.getRecentFiles` with `this._recentFiles` that we memorized during the last calls to `_buildAndSendMenuTemplate`. If it's different, we call `_buildAndSendMenuTemplate`.

The menu will be updated (and this._recentFiles updated). Done! :)